### PR TITLE
feat: Adds invisible character detection logic and updates tests

### PR DESCRIPTION
**Pull request by CristopherAguilar10**
This PR will add in invisible character detection functionality in code-expressions to the Cairo plugin.

**Invisible Character and Whitespaces detection:**

- The Function `check_invisible_characters` is implemented and performs the following checks in the `Expr`: 
- Whitespaces: a warning of (`Severity::Warning`) if present.
- Invisible characters: error (`Severity::Error`) on C-style  string literals like` \u{200B}`, `\u{00AD}`, `\u{200C}`, `\u{200D}`, `\u{202C}`, `\u{FEFF}`.

**Motivation:**

- The invisible kind of characters, like `zero-width space`, can create bugs that are impossible to trace in the code. This change enables the detection of such unwanted characters and possibly a warning/blocking their usage in code to guarantee the quality of source code.

**Diagnostic Messages Add following two diagnostic messages:**

- "Whitespace detected.": Whenever whitespace is detected.
- "Invisible character detected.": When invisible characters are detected.

### CODE REVIEW

**Comment by 0xLucqs** [comment]
check the test file it's full of unrelated diagnostics

**Comment by CristopherAguilar10** [comment]
@0xLucqs, could you check the changes, thanks  🚀

File: crates/cairo-lint-core/src/plugin.rs [hunk 1]
```diff
@@ -142,6 +144,14 @@ impl AnalyzerPlugin for CairoLint {
                             &mut diags,
                         );
                     }
+                    SyntaxKind::ExprInlineMacro => {
+                        invisible_characters::check_invisible_characters(
+                            &AstExpr::from_syntax_node(db.upcast(), node),  
+                            db.upcast(),                                    
+                            &mut diags                                      
+                        );
+                    }
```
Comment (0xLucqs) [inline crates/cairo-lint-core/src/plugin.rs:153]: you should check the whole file not just inline macros

File: crates/cairo-lint-core/tests/test_files/invisible_characters/invisible_characters [hunk 2]
```diff
@@ -0,0 +1,165 @@
+//! > detect byte order mark
+
+//! > cairo_code
```
Comment (0xLucqs) [inline crates/cairo-lint-core/tests/test_files/invisible_characters/invisible_characters:3]: remove every diagnostic that's not related to this lint

**Review by 0xLucqs — CHANGES_REQUESTED** [review]

**Pull request closed without merging**